backend/jobsearch/src/main/kotlin/com/jobsearch/service/VacancyService.kt

```
@@ -1,87 +1,111 @@
+    @Transactional
+    fun updateVacancy(vacancyId: Int, vacancyDto: VacancyRequestDTO): VacancyResponseDTO {
 
-        val selectedJobFamily = jobFamilyService.findByJobFamilyId(vacancyDto.jobFamilyId!!)
-            .orElseThrow { NoSuchElementException("No vacancy found with id ${vacancyDto.jobFamilyId}") }
+        val vacancy = vacancyRepository.findById(vacancyId)
+            .orElseThrow { NotFoundException("No vacancy found with id $vacancyId") }
+        val manager = userService.retrieveAuthenticatedUser()
+        if (vacancy.manager.email != manager.email ) throw ForbiddenException("You are not allowed to edit this vacancy.")
 
+        val selectedJobFamily = jobFamilyService.findByJobFamilyId(vacancyDto.jobFamilyId)
         vacancy.name = vacancyDto.name
         vacancy.companyName = vacancyDto.companyName
         vacancy.salaryExpectation = vacancyDto.salaryExpectation
         vacancy.description = vacancyDto.description
         vacancy.jobFamily = selectedJobFamily
 
         val updatedVacancy = vacancyRepository.save(vacancy)
-        return mapToVacancyDto(updatedVacancy)
-    }
-
-    fun deleteVacancy(vacancyId: Int): String {
-        val vacancy = vacancyRepository.findById(vacancyId)
-            .orElseThrow { NoSuchElementException("No vacancy found with id $vacancyId") }
 
-        vacancyRepository.delete(vacancy)
-        return "Vacancy deleted successfully"
+        return mapToVacancyResponseDto(updatedVacancy)
     }
 
     @Transactional
-    fun findVacanciesByFilter(salary: Int?, jobFamilyId: Int?, yearsOfExperience: Int?): List<VacancyDto> {
-        val vacancies = vacancyRepository.findVacanciesByFilters(salary, jobFamilyId, yearsOfExperience)
-        return vacancies.map {
-            mapToVacancyDto(it)
-        }
-    }
-
-    fun mapToVacancyDto(vacancy: Vacancy): VacancyDto {
+    fun deleteVacancy(vacancyId: Int) {
+        val vacancy = vacancyRepository.findById(vacancyId)
+            .getOrElse { return }
```

if we cannot find a vacancy by `id` wu should throw not found exception